packages/aws-rfdk/lib/deadline/lib/configure-spot-event-plugin.ts

```diff
@@ -0,0 +1,652 @@
+ *   environments.
+ * - The AWS Lambda function that is created by this resource has access to both the certificates used to connect to the render queue,
+ *   and the render queue port. An attacker that can find a way to modify and execute this lambda could use it to
+ *   execute any requets against the render queue. You should not grant any additional actors/principals the ability to modify
+ *   or execute this Lambda.
+ */
+export class ConfigureSpotEventPlugin extends Construct {
+
+  /**
+   * Only one Spot Event Plugin Configuration is allowed per render queue / repository.
+   */
+  private static uniqueRenderQueues: Set<IRenderQueue> = new Set<IRenderQueue>();
+
+  constructor(scope: Construct, id: string, props: ConfigureSpotEventPluginProps) {
+    super(scope, id);
+
+    if (ConfigureSpotEventPlugin.uniqueRenderQueues.has(props.renderQueue)) {
+      throw new Error('Only one ConfigureSpotEventPlugin construct is allowed per render queue.');
+    }
+    else {
+      ConfigureSpotEventPlugin.uniqueRenderQueues.add(props.renderQueue);
+    }
+
+    if (props.renderQueue instanceof RenderQueue) {
+      // We do not check the patch version, so it's set to 0.
+      const minimumVersion: Version = new Version([10, 1, 12, 0]);
+
+      if (props.renderQueue.version.isLessThan(minimumVersion)) {
+        throw new Error(`Minimum supported Deadline version for ${this.constructor.name} is ` +
+        `${minimumVersion.versionString}. ` +
+        `Received: ${props.version.versionString}.`);
+      }
+
+      if (props.spotFleets && props.spotFleets.length !== 0) {
+        props.renderQueue.addSEPPolicies(props.configuration?.enableResourceTracker ?? true);
+
+        const fleetRoles = props.spotFleets.map(sf => sf.fleetRole.roleArn);
+        const fleetInstanceRoles = props.spotFleets.map(sf => sf.fleetInstanceRole.roleArn);
+        new Policy(this, 'SpotEventPluginPolicy', {
+          statements: [
+            new PolicyStatement({
+              actions: [
+                'iam:PassRole',
+              ],
+              resources: [...fleetRoles, ...fleetInstanceRoles],
+              conditions: {
+                StringLike: {
+                  'iam:PassedToService': 'ec2.amazonaws.com',
+                },
+              },
+            }),
+            new PolicyStatement({
+              actions: [
+                'ec2:CreateTags',
+              ],
+              resources: ['arn:aws:ec2:*:*:spot-fleet-request/*'],
+            }),
+          ],
+          roles: [
+            props.renderQueue.grantPrincipal as Role,
+          ],
+        });
+      }
+    }
+    else {
+      throw new Error('The provided render queue is not an instance of RenderQueue class. Some functionality is not supported.');
+    }
+
+    const region = Construct.isConstruct(props.renderQueue) ? Stack.of(props.renderQueue).region : Stack.of(this).region;
```

Comment:
Dependending on the comment above, I would actually prefer to simplify and remove `Construct.isConstruct`, as we know that our `RenderQueue` is a construct and we don't support other RenderQueues at the moment.